File: sapi_frontend/src/types/document.ts
export type DocumentStatus = 'UPLOADED' | 'PROCESSING' | 'PROCESSED' | 'REVIEW_NEEDED' | 'ERROR';

export interface DocumentType {
  id: string;
  name: string;
  description?: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ExtractedField {
  field_name: string;
  field_label?: string;
  ai_extracted_value?: string;
  ai_confidence?: string;
  final_value: string;
  is_corrected: boolean;
  corrected_at?: string;
}

export interface Document {
  id: string;
  original_filename: string;
  status: DocumentStatus;
  classification_confidence?: string;
  upload_user_id: string;
  document_type_id?: string;
  document_type_name?: string;
  processing_started_at?: string;
  processing_completed_at?: string;
  created_at: string;
  updated_at: string;
}

export interface DocumentDetail extends Document {
  file_size?: string;
  mime_type?: string;
  document_type?: DocumentType;
  executive_summary?: string;
  extracted_fields: ExtractedField[];
  processing_error?: string;
}

export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

export interface DocumentStatusResponse {
  id: string;
  status: DocumentStatus;
  message: string;
}

export interface ExtractedDataUpdate {
  field_name: string;
  new_value: string;
}

export interface UploadResponse {
  id: string;
  status: DocumentStatus;
  message: string;
}

File: sapi_backend/app/schemas/document.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID
from datetime import datetime
from typing import Optional, List
from enum import Enum


class DocumentStatus(str, Enum):
    UPLOADED = "UPLOADED"
    PROCESSING = "PROCESSING"
    PROCESSED = "PROCESSED"
    REVIEW_NEEDED = "REVIEW_NEEDED"
    ERROR = "ERROR"


class DocumentTypeBase(BaseModel):
    name: str
    description: Optional[str] = None


class DocumentTypeCreate(DocumentTypeBase):
    pass


class DocumentTypeResponse(DocumentTypeBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    is_active: bool
    created_at: datetime
    updated_at: datetime


class DocumentBase(BaseModel):
    original_filename: str


class DocumentCreate(DocumentBase):
    document_type_id: Optional[UUID] = None


class DocumentUpdate(BaseModel):
    status: Optional[DocumentStatus] = None
    document_type_id: Optional[UUID] = None
    classification_confidence: Optional[str] = None
    executive_summary: Optional[str] = None
    processing_started_at: Optional[datetime] = None
    processing_completed_at: Optional[datetime] = None
    processing_error: Optional[str] = None


class ExtractedFieldResponse(BaseModel):
    field_name: str
    field_label: Optional[str] = None
    ai_extracted_value: Optional[str] = None
    ai_confidence: Optional[str] = None
    final_value: str
    is_corrected: bool
    corrected_at: Optional[datetime] = None

class DocumentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    original_filename: str
    status: DocumentStatus
    classification_confidence: Optional[str] = None
    upload_user_id: UUID
    document_type_id: Optional[UUID] = None
    processing_started_at: Optional[datetime] = None
    processing_completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime


class DocumentListResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    original_filename: str
    status: DocumentStatus
    document_type_name: Optional[str] = None
    classification_confidence: Optional[str] = None
    created_at: datetime


class DocumentStatusResponse(BaseModel):
    id: UUID
    status: DocumentStatus
    message: str


class DocumentDetailResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    original_filename: str
    file_size: Optional[str] = None
    mime_type: Optional[str] = None
    status: DocumentStatus
    document_type: Optional[DocumentTypeResponse] = None
    classification_confidence: Optional[str] = None
    executive_summary: Optional[str] = None
    extracted_fields: List[ExtractedFieldResponse] = []
    processing_started_at: Optional[datetime] = None
    processing_completed_at: Optional[datetime] = None
    processing_error: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class ExtractedDataUpdate(BaseModel):
    field_name: str
    new_value: str


class ExtractedDataUpdateList(BaseModel):
    updates: List[ExtractedDataUpdate]
